src/controllers/services.controllers.js

```diff
@@ -1,26 +1,154 @@
-import { dataSource } from "../../db.js"
 
-export const createNewCourse = async(req,res)=>{
-    console.log(req.body)
+import { dataSource } from "../../db.js";
+import { isValidType } from "../helpers/validateServiceType.js";
+import { uploadFile } from "../services/cloudinary.js";
+import { createRequire } from "module";
+const require = createRequire(import.meta.url);
+const {IsNull} = require("typeorm");
+const formationServiceRepository = dataSource.getRepository("FormationServices")
+
+
+export const createNewFormationService = async (req, res) => {
+
+  const type = req.body.type
+  if(!type || type && !isValidType(type)){
+    return res
+    .status(400)
+    .json({ isSuccess: false, message: "Please provide a valid type of course: 'course','seminar', 'workshop', 'diploma'" });
+  }
+  
+  
+  try {
     
-    const courseRepository = dataSource.getRepository("Course")
     let data = {
-        name:"Hola",
-        general_info:"Hola",
-        syllabus:"",
-        hours:20,
-        exhibitor_name:"Juan Carlos Montes",
-        exhibitor_photo:"www",
-        organized_by:"sadsad",
-        supported_by_name:"PYGLO",
-        supported_by_photo:"www",
-        cost:2500000,
-        discounts:"30% Egresados",
-        inscription_url:"www"
+      name: req.body.name,
+      general_info: req.body.general_info,
+      type:req.body.type,
+      image: req.body.image,
+      syllabus: req.body.syllabus,
+      hours: req.body.hours,
+      exhibitor_name: req.body.exhibitor_name,
+      organized_by: req.body.organized_by,
+      supported_by_name: req.body.supported_by_name,
+      cost: req.body.cost,
+      discounts: req.body.discounts,
+      inscription_url: req.body.inscription_url,
```

Comment:
question: I don't understand how you are saving the images' URLs, I can see you are receiving it from the body, but it would be determined before uploading it to Cloudinary